odoo/tests/common.py

```diff
@@ -587,6 +587,31 @@ def assertTreesEqual(self, n1, n2, msg=None):
             self.assertTreesEqual(c1, c2, msg)
 
 
+    @contextmanager
+    def capture_trigger(self, cron_id=None):
+        """
+        Context manager that capture new triggers in the ``triggers``
+        attribute of the context object.
+
+        :param cron_id: An optional cron record id (int) or xmlid (str)
+                        to only capture triggers for that cron.
+        """
+        if isinstance(cron_id, str):  # xmlid case
+            cron_id = self.env.ref(cron_id).id
+        domain = [('cron_id', '=', cron_id)] if cron_id else []
+        old_triggers = self.env['ir.cron.trigger'].sudo().search(domain)
+        triggers = None
+
+        class capture:
+            @property
+            def triggers(self):
+                return triggers or old_triggers.search(domain) - old_triggers
+
+        c = capture()
+        yield c
+        triggers = c.triggers
```

Comment:
This is very obscure, you kind of force the property.
Why not an obvious assignment?
```suggestion
        c = capture()
        yield c
        c.triggers = old_triggers.search(domain) - old_triggers
```

Note that you could return a plain stupid list:
```suggestion
        result = []
        yield result
        result.extend(old_triggers.search(domain) - old_triggers)
```

Reply:
Technically speaking, you could return the recordset itself, but it's a bit hackish:
```python
result = old_triggers.browse()
return result
result._ids = result._prefetch_ids = (old_triggers.search(domain) - old_triggers)._ids
```